### PR TITLE
Add section on constructors, static members and member invocations.

### LINKED ISSUE #21362: Clarify constructors
Hello @cartermp,

I'm wondering what the style guide says about constructors and long ones in particular.

For example:
```fsharp
let thing = new Foobar(a, b)

let otherThing =
    new Foobar(
        longname1,
        longname2
    )

let m = new Meh(a)

let otherThing =
    new Foobar(
        longname1,
        longname2,
        longname3,
        longname4,
        longname5,
        longname6,
        longname7
    )
```

When to use multiple lines? Where to align in the indentation?

Fantomas today would format this as:
```fsharp
let thing = new Foobar(a, b)

let otherThing =
    new Foobar(longname1, longname2)

let m = new Meh(a)

let otherThing =
    new Foobar(longname1,
               longname2,
               longname3,
               longname4,
               longname5,
               longname6,
               longname7)
```

I'm asking this question in order to see if the Microsoft style guide will deviate from what [the G-Research guide has in mind](https://github.com/G-Research/fsharp-formatting-conventions/pull/11).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8c09996f-24ac-cf39-ef43-d7909cbc2b89
* Version Independent ID: 1efdd523-8552-6421-e149-74f76826ef8f
* Content: [F# code formatting guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting)
* Content Source: [docs/fsharp/style-guide/formatting.md](https://github.com/dotnet/docs/blob/master/docs/fsharp/style-guide/formatting.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **phcart**

### PR DESCRIPTION
**Pull request by nojaf**
## Summary

Add a new section to the F# style guide on constructors, static members and member invocations.

Fixes #21362

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -884,6 +884,33 @@ let makeStreamReader x = new System.IO.StreamReader(path=x)
...
+If the expression is long, use newlines and indent one scope, rather than indenting to the bracket.
+
+```fsharp
+let person =
+    new Person(
+        argument1,
+        argument2
+    )
+
+let p = new Person(a, b) // keep in one line if short
```
Comment (cartermp) [inline docs/fsharp/style-guide/formatting.md:898]: This should be pulled into its own section, like so:

```
If the expression for calling a constructor is short, keep it on one line:

<example>
```

Then the bit about longer expressions can come afterwards.

This way we don't need the user to look at a code comment to see the guideline.

**Review by cartermp — CHANGES_REQUESTED** [review]
Mostly great, just think that the one-line bit can be pulled out into its own sentence + sample that comes before the long expressions bit,

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md]: Good point.

**Review by cartermp — APPROVED** [review]
Thanks! Looks great.

**Comment by nojaf** [comment]
Out of curiosity, how does this get published?
Did I need to bump the `ms.date` field at the top?

**Comment by cartermp** [comment]
Oh, bumping the data i think helps with SEO but that's about it. these get pulled into a shipping branch every weekday if I'm not mistaken, changes go live in about an hour. Personally I think it's a little weird that we gotta do that, but I'm sure there are reasons for it

**Pull request merged**